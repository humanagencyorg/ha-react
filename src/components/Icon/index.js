import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ src, className, inline, ref }) => (
  // Vova: unset max-width: none; for img tag
  // because on MultiSelectOptionButton icon hidden
  // when text overflowed is active and max-width: 100%;
  inline    ?
    <img src={src} style={{maxWidth: 'none'}} />    :
    <div className={className} ref={ref}>
      <img src={src} />
    </div>
);

Icon.propTypes = {
  src: PropTypes.string,
  className: PropTypes.string,
  inline: PropTypes.bool,
};

Icon.defaultProps = {
  inline: false,
};

export default Icon;
