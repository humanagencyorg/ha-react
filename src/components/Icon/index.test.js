import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './index';

describe('Icon component', () => {
  it('matches snapshot', () => {
    const component = renderer
      .create(<Icon src="http://default.props/image.png"/>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
