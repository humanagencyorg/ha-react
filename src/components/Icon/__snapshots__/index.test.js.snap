// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component matches snapshot 1`] = `
<div>
  <img
    src="http://default.props/image.png"
  />
</div>
`;

exports[`Icon component render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "src": "http://default.props/image.png",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "src": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
